### ISSUE TITLE
asin() acos() crash on system

### ISSUE THREAD

**Issue by qufaud99** [godotengine/godot#8925]
**Operating system or device - Godot version:**
MacBook Pro (Retina, 13-inch, Early 2015)

**Issue description:**

these make godot editor quit unexpectedly

**Steps to reproduce:**

**Link to minimal example project:**

Just create simple codes

extends Node2D

func_ready():
     var simple=asin(10.0)

      
OR

extends Node2D

func _ready():
    var simple=acos(10.0)

**Comment by volzhs** [comment]
what is your godot version?
I tested with follow code and could not reproduce it with godot 2.1.3
```gdscript
func _ready():
	var simple1=asin(10.0)
	var simple2=acos(10.0)
	prints(simple1)  # prints -nan
	prints(simple2)  # prints -nan
```

**Comment by qufaud99** [comment]
godot version I tested was v2.1.3.stable.official

**Comment by vnen** [comment]
I can reproduce on WIndows 10.

**Comment by vnen** [comment]
Got a null dereference (`this` is `NULL`) with this stack trace (Visual Studio is not really good at stack traces):

```
 	Variant::clear() Line 1107
 	Variant::reference(const Variant & p_variant) Line 930
 	Variant::operator=(const Variant & p_variant) Line 2567
 	GDCompiler::_parse_function(GDScript * p_script, const GDParser::ClassNode * p_class, const GDParser::FunctionNode * p_func, bool p_for_ready) Line 1336
 	GDCompiler::_parse_class(GDScript * p_script, GDScript * p_owner, const GDParser::ClassNode * p_class, bool p_keep_state) Line 1707
 	GDCompiler::compile(const GDParser * p_parser, GDScript * p_script, bool p_keep_state) Line 1841
 	GDScript::reload(bool p_keep_state) Line 513
 	GDScriptLanguage::reload_tool_script(const Ref<Script> & p_script, bool p_soft_reload) Line 1561
 	ResourceFormatSaverGDScript::save(const String & p_path, const Ref<Resource> & p_resource, unsigned int p_flags) Line 1820
 	ResourceSaver::save(const String & p_path, const Ref<Resource> & p_resource, unsigned int p_flags) Line 73
 	EditorNode::save_resource_in_path(const Ref<Resource> & p_resource, const String & p_path) Line 546
 	EditorNode::save_resource(const Ref<Resource> & p_resource) Line 562
	ScriptEditor::save_all_scripts() Line 2135
 	ScriptEditorPlugin::save_external_data() Line 2903
 	EditorData::save_editor_external_data() Line 371
 	EditorNode::_save_scene(String p_file, int idx) Line 976
 	EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) Line 2061
 	EditorNode::_menu_option(int p_option) Line 627
 	EditorNode::_run(bool p_current, const String & p_custom) Line 1815
 	EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) Line 2563
 	EditorNode::_menu_option(int p_option) Line 627
 	MethodBind1<EditorNode,int>::call(Object * p_object, const Variant * * p_args, int p_arg_count, Variant::CallError & r_error) Line 730
 	Object::call(const StringName & p_method, const Variant * * p_args, int p_argcount, Variant::CallError & r_error) Line 853
 	Object::emit_signal(const StringName & p_name, const Variant * * p_args, int p_argcount) Line 1156
 	Object::emit_signal(const StringName & p_name, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5) Line 1197
 	BaseButton::_unhandled_input(InputEvent p_event) Line 435
 	MethodBind1<BaseButton,InputEvent>::call(Object * p_object, const Variant * * p_args, int p_arg_count, Variant::CallError & r_error) Line 730
 	Object::call_multilevel(const StringName & p_method, const Variant * * p_args, int p_argcount) Line 594
 	SceneTree::_call_input_pause(const StringName & p_group, const StringName & p_method, const InputEvent & p_input) Line 826
 	Viewport::unhandled_input(const InputEvent & p_event) Line 2280
 	Viewport::_vp_unhandled_input(const InputEvent & p_ev) Line 1392
 	MethodBind1<Viewport,InputEvent const & __ptr64>::call(Object * p_object, const Variant * * p_args, int p_arg_count, Variant::CallError & r_error) Line 730
 	Object::call(const StringName & p_method, const Variant * * p_args, int p_argcount, Variant::CallError & r_error) Line 853
 	Object::call(const StringName & p_name, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5) Line 733
 	SceneTree::call_group(unsigned int p_call_flags, const StringName & p_group, const StringName & p_function, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5) Line 208
 	SceneTree::input_event(const InputEvent & p_event) Line 447
 	InputDefault::parse_input_event(const InputEvent & p_event) Line 350
 	OS_Windows::process_key_events() Line 826
 	OS_Windows::process_events() Line 1834
 	OS_Windows::run() Line 2190
 	widechar_main(int argc, wchar_t * * argv) Line 144
 	main(int _argc, char * * _argv) Line 169
```

It does not always crash and when it doesn't, I get this error on console:

```
ERROR: HashMap<class Variant,int,struct VariantHasher,3,8>::next: Condition ' !e ' is true. returned: 0
   At: core\hash_map.h:507
```

**Comment by qufaud99** [comment]
perhaps is it an issue?

**Comment by bojidar-bg** [comment]
Might be related to #7354 which was fixed on master by #7815 and #8393.
@vnen Which version did you test?

**Comment by vnen** [comment]
@bojidar-bg might be related, yes. I'm testing on 2.1.3.

**Comment by avril-gh** [comment]
Windows 7 32bit, godot 2.2

confirming - it crashes for following code

```
func _ready():
	var simple1=asin(10.0)
	var simple2=acos(10.0)
	prints(simple1)  # prints -nan
	prints(simple2)  # prints -nan
```

**Comment by volzhs** [comment]
then probably it has been fixed as @bojidar-bg commented?
I forgot to mention that I'm using 2.1.3 custom build with almost latest commit.

**Comment by vnen** [comment]
Did #7815 get cherry-picked to the 2.1 branch?

**Comment by volzhs** [comment]
@vnen I did quick search but both #7815 and #8393 are not on 2.1 branch.
maybe it's fine only x11?

**Comment by eon-s** [comment]
2.1.3 stable on Ubuntu 17.04, no crash.

**Comment by akien-mga** [comment]
Fixed by #10490.
